Dockerfile.test

```diff
@@ -9,6 +9,7 @@ WORKDIR /root
 RUN python3 -m venv venv
 RUN --mount=type=cache,target=/root/.cache /bin/bash -c "source venv/bin/activate && \
     pip install -r requirements.test.txt -r requirements.dev.txt"
+RUN python3 -m spacy download en_core_web_sm
```

Comment:
Yet again, this PR surfaced another bug in the notebooks. Here we were missing this dependency in the test!